google/auth/pluggable.py

```diff
@@ -283,18 +283,20 @@ def revoke(self, request):
         response = json.loads(result.stdout.decode("utf-8"))
         self._validate_revoke_response(response)
 
-    @property
-    def external_account_id(self):
-        """Returns the external account identifier.
-
-        When service account impersonation is used the identifier is the service
-        account email.
+    def external_account_id(self, request):
```

Comment:
I don't think we should expose this here. Should be in external account, and private?

Reply:
Not sure I am understanding correctly as this is just calling introspect_token which can only be invoked from gcloud when using these credentials

Reply:
> Not sure I am understanding correctly as this is just calling introspect_token which can only be invoked from gcloud when using these credentials

I think my understanding is different. We are open to any client (gcloud or 3rd party) that has basic auth to introspect the token they acquired. The introspect of token can be done for external_account type. And for pluggable auth specifically used for gcloud. Retrieving external account id is just one usage of that and we do that in the `pluggable_auth.Crentials` privately.